### ISSUE TITLE
PRs to fix patches in *cloud_controller_ng* upstream project

### ISSUE THREAD

**Issue by fargozhu** [cloudfoundry-incubator/kubecf#504]
./api/cloud_controller_ng/jobs/patch_post-start.sh
 ./api/cloud_controller_ng/jobs/remove_tee_output_to_sys_log.sh
 ./api/cloud_controller_ng/bpm/patch_bpm.sh

**Comment by fargozhu** [comment]
relates with #12

**Comment by andreas-kupries** [comment]
Main directory: `bosh/releases/pre_render_scripts/api/cloud_controller_ng`

Affects release [capi-release 1.89.0](https://github.com/cloudfoundry/capi-release/commit/c3af0a025b66e0d316110698a6bbb40766fcad6a)

|Patch|Notes|
|---|---|
|[bpm/patch_bpm.sh](https://github.com/cloudfoundry-incubator/kubecf/blob/master/bosh/releases/pre_render_scripts/api/cloud_controller_ng/bpm/patch_bpm.sh)|Looks to be a Suse specific. Might be possible to make it configurable.|
|[jobs/patch_cc_uploader_ctl.sh](https://github.com/cloudfoundry-incubator/kubecf/blob/master/bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cc_uploader_ctl.sh)|Make ulimit on fds configurable. See also #505. (`rep_ctl` patch, diego-release)|
|[jobs/patch_cloud_controller_api_ctl.sh](https://github.com/cloudfoundry-incubator/kubecf/blob/master/bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_cloud_controller_api_ctl.sh)|Ditto.|
|[jobs/patch_file_server_ctl.sh](https://github.com/cloudfoundry-incubator/kubecf/blob/master/bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_file_server_ctl.sh)|Ditto.|
|[jobs/patch_post-start.sh](https://github.com/cloudfoundry-incubator/kubecf/blob/master/bosh/releases/pre_render_scripts/api/cloud_controller_ng/jobs/patch_post-start.sh)|Fix permissions on CC lgo file.|

**Comment by andreas-kupries** [comment]
@viovanov @f0rmiga @mook-as @jandubois @fargozhu @prabalsharma @jimmykarily @thardeck @svollath @viccuad Ping. Above the table listing our local patches in the cc-ng area and the commits proposed to become the equivalent upstream PRs.

Note that some of the changes relate to changes in the diego-cell area, regarding the handling of fd ulimits. 

Wrt the bpm patch I feel that a bit of discussion is needed. Part of it might be bugfix (nginx maintenance, maybe DYNO), part is our config (no newrelic active).

**Comment by viovanov** [comment]
@andreas-kupries yes, for bpm let's hide the changes behind configuration values.
The dyno uses a spec.index in bpm, which we don't support. 
Maybe a `export_dyno` boolean?

**Comment by andreas-kupries** [comment]
Made the removed parts configurable.

**Comment by andreas-kupries** [comment]
The cc-ng post-start patch has been found superfluous. PR #863 removes it. The PR https://github.com/cloudfoundry/capi-release/pull/171 will be retracted and closed.

**Comment by andreas-kupries** [comment]
All the ulimit patches were found superfluous. PR #1040 removed them. The PR cloudfoundry/capi-release#172 has been retracted and closed.

The remaining patches for this ticket will be re-checked regarding necessity, and different solutions.

**Comment by andreas-kupries** [comment]
The PR https://github.com/cloudfoundry/capi-release/pull/170 is still pending upstream. no comments nor actions from upstream.

**Comment by andreas-kupries** [comment]
Based on https://github.com/cloudfoundry-incubator/kubecf/issues/520#issuecomment-648407686

Since this is filed upstream now, will close. Will track the upstream PR so hopefully, when it's done, we can somehow track it to what we need to consume next.
